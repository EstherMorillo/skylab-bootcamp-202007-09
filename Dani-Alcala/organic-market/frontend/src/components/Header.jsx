import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import "./Header.css";
import productStore from "../stores/productStore";
// import TextInput from "./TextInput";

function Header() {
  // const [productName, setProductName] = useState("");
  const [showForm, setShowForm] = useState(false);
  const [genre, setGenre] = useState("");
  const [numberCart, setNumberCart] = useState(productStore.getNumberCart());

  setInterval(() => setNumberCart(productStore.getNumberCart()), 100);

  useEffect(() => {
    productStore.addChangeListener(onChange);
    return () => productStore.removeChangeListener(onChange);
  },[])

  function handleClick() {
    setShowForm(true);
  }

  function onChange(){
    setNumberCart(productStore.getNumberCart()); 
  }
  
  return (
    <>
      <section className="main-nav">
        

        <Link to="/" className="icon__logo"></Link>

        <div>

          <p className="header__logo-text">ORGANIC MARKET</p>
          
        </div>

        <div className="hidden"></div>

        {/* {searchForm} */}

        <Link to={handleClick} className="icon__search"></Link>

        <Link to="/login" className="icon__login"></Link>
        
        <Link to="/cart" className="icon__bag">
          <div className="icon__number">{numberCart}</div>
        </Link>
      
      </section>
    </>
  );
}

export default Header;
