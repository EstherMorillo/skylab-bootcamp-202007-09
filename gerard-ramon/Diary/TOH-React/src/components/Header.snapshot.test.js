import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { BrowserRouter as Router } from 'react-router-dom';

describe('header snapshot', () => {
	it('Should match', () => {
		const tree = renderer.create(
			<Router>
				<Header />
			</Router>
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
