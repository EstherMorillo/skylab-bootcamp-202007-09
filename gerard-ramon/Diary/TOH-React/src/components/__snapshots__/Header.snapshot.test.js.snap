// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header snapshot Should match 1`] = `
<header>
  <h1>
    Hero Tour
  </h1>
  <a
    aria-current="page"
    className="button active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Dashboard
  </a>
  <a
    aria-current={null}
    className="button"
    href="/heroes"
    onClick={[Function]}
  >
    Heroes
  </a>
</header>
`;
